UnicornRender/video/source/Primitives.cpp

```diff
@@ -128,9 +138,131 @@ Mesh& Primitives::Sphere(Mesh& mesh, float radius, uint32_t rings, uint32_t sect
+    for (unsigned int i = 0; i < mat->GetTextureCount(type); ++i)
+    {
+        aiString str;
+        mat->GetTexture(type, i, &str);
+        std::string path = str.C_Str();
+        std::replace(path.begin(), path.end(), '\\', '/');
+        textures.emplace_back(directory + "/" + path);
+    }
+    return textures;
+}
+
+Mesh* ProcessMesh(aiMesh* mesh, aiScene const* scene, std::string const& dir)
+{
+    std::vector<uint32_t> indices;
+    std::vector<Vertex> vertices;
+
+    for (unsigned int i = 0; i < mesh->mNumVertices; i++)
+    {
+        Vertex vertex;
+
+        glm::vec3 vector;
+        vector.x = mesh->mVertices[i].x;
+        vector.y = mesh->mVertices[i].y;
+        vector.z = mesh->mVertices[i].z;
+        vertex.pos = vector;
+
+        if (mesh->mTextureCoords[0]) // does the mesh contain texture coordinates?
+        {
+            glm::vec2 vec;
+            vec.x = mesh->mTextureCoords[0][i].x;
+            vec.y = mesh->mTextureCoords[0][i].y;
+            vertex.tc = vec;
+        }
+        else
+        {
+            vertex.tc = glm::vec2(0.0f, 0.0f);
+        }
+
+        vertices.push_back(vertex);
+    }
+
+    for (unsigned int i = 0; i < mesh->mNumFaces; i++)
+    {
+        aiFace face = mesh->mFaces[i];
+        for (unsigned int j = 0; j < face.mNumIndices; j++)
+        {
+            indices.push_back(face.mIndices[j]);
+        }
+    }
+
+    Mesh* unicornMesh = new Mesh;
+
+    unicornMesh->name = mesh->mName.C_Str();
+
+    auto mat = std::make_shared<Material>();
+    
+    aiMaterial *material = scene->mMaterials[mesh->mMaterialIndex];
+    unicornMesh->m_diffuse = LoadMaterialTextures(material, aiTextureType_DIFFUSE, dir);
+    unicornMesh->m_normal = LoadMaterialTextures(material, aiTextureType_NORMALS, dir);
+    unicornMesh->m_aoMaps = LoadMaterialTextures(material, aiTextureType_LIGHTMAP, dir);
+    unicornMesh->m_emissive = LoadMaterialTextures(material, aiTextureType_EMISSIVE, dir);
+    unicornMesh->m_metalRougness = LoadMaterialTextures(material, aiTextureType_UNKNOWN, dir);
+    aiColor3D color(0.f, 0.f, 0.f);
+    material->Get(AI_MATKEY_COLOR_DIFFUSE, color);
+    mat->SetColor({ color.r, color.g, color.b });
+
+    if (unicornMesh->m_diffuse.size() > 0)
+    {
+        auto diffuseTex = std::make_shared<Texture>();
+        diffuseTex->Load(unicornMesh->m_diffuse.at(0));
+        mat->SetAlbedo(diffuseTex);
+    }
+    unicornMesh->SetMaterial(mat);
+
+    unicornMesh->SetMeshData(vertices, indices);
+
+    return unicornMesh;
+}
+
+std::list<Mesh*> Primitives::LoadModel(std::string const& path)
+{
+    std::list<Mesh*> meshes;
+
+    Assimp::Importer importer;
+
+    aiScene const* scene = importer.ReadFile(path,
+        aiProcess_Triangulate |
+        aiProcess_FlipUVs);
+
+    if (!scene || scene->mFlags & AI_SCENE_FLAGS_INCOMPLETE || !scene->mRootNode)
+    {
+        spdlog::get(utility::Logger::LoggerName())->error("ERROR importing %s mesh : %s", path.c_str(),
```

Comment:
`std::shared_ptr<spdlog::logger>` should be a class variable. There is no need to call `get()` every time.